G4HepEm/G4HepEmData/CMakeLists.txt

```diff
@@ -20,30 +20,15 @@ set(G4HEPEMDATA_CXX_sources
   src/G4HepEmSBTableData.cc
 )
 
-if(BUILD_STATIC_LIBS)
-  add_library(g4HepEmData STATIC ${G4HEPEMDATA_CXX_sources})
-else()
-  add_library(g4HepEmData SHARED ${G4HEPEMDATA_CXX_sources})
-endif()
-
-add_library(${PROJECT_NAME}::g4HepEmData ALIAS g4HepEmData)
-
-target_compile_definitions(g4HepEmData PUBLIC $<$<BOOL:${G4HepEm_CUDA_BUILD}>:G4HepEm_CUDA_BUILD>)
-
-target_compile_features(g4HepEmData PUBLIC cxx_std_${CMAKE_CXX_STANDARD})
+g4hepem_add_library(g4HepEmData SOURCES ${G4HEPEMDATA_CXX_sources} HEADERS ${G4HEPEMDATA_headers})
 
-target_include_directories(g4HepEmData PUBLIC
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}>)
-
- # TODO: Determine when/if to use CUDA::cudart_static
-target_link_libraries(g4HepEmData PRIVATE $<$<BOOL:${G4HepEm_CUDA_BUILD}>:CUDA::cudart>)
+if(BUILD_SHARED_LIBS)
+  target_compile_definitions(g4HepEmData PUBLIC $<$<BOOL:${G4HepEm_CUDA_BUILD}>:G4HepEm_CUDA_BUILD>)
+  target_link_libraries(g4HepEmData PRIVATE $<$<BOOL:${G4HepEm_CUDA_BUILD}>:CUDA::cudart>)
+endif()
 
-## ----------------------------------------------------------------------------
-## Install G4HepEm libraries and headers
-install(FILES ${G4HEPEMDATA_headers} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}")
-install(TARGETS g4HepEmData
-  EXPORT ${PROJECT_NAME}Targets
-  ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-  LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
+if(BUILD_STATIC_LIBS)
+  target_compile_definitions(g4HepEmData-static PUBLIC $<$<BOOL:${G4HepEm_CUDA_BUILD}>:G4HepEm_CUDA_BUILD>)
+  # TODO: Determine when/if to use CUDA::cudart_static
+  target_link_libraries(g4HepEmData-static PRIVATE $<$<BOOL:${G4HepEm_CUDA_BUILD}>:CUDA::cudart>)
```

Comment:
I _think_ using the static variant is _probably_ o.k. here if only to state "we assume if you're linking to G4HepEM statically, then you want everything static". Almost certainly some corner cases for upstream though...

Reply:
Yes, it's tricky because, IIRC, CUDA doesn't like at all being linked dynamically and statically at the same time. Would probably need some testing what's the best here, but I'm not even sure we ever tried a static build of G4HepEm with CUDA enabled... I'll probably leave this as is for now, it's not worse than before.